osu.Game/Skinning/LocalSkinOverrideContainer.cs

```diff
@@ -45,20 +45,20 @@ public SampleChannel GetSample(string sampleName)
 
         public TValue? GetValue<TConfiguration, TValue>(Func<TConfiguration, TValue?> query) where TConfiguration : SkinConfiguration where TValue : struct
         {
-            TValue? val = null;
+            TValue? val;
             if ((source as Skin)?.Configuration is TConfiguration conf)
-                val = query?.Invoke(conf);
-
-            return val ?? fallbackSource?.GetValue(query);
+                if (beatmapSkins && (val = query?.Invoke(conf)) != null)
+                    return val;
+            return fallbackSource?.GetValue(query);
         }
 
         public TValue GetValue<TConfiguration, TValue>(Func<TConfiguration, TValue> query) where TConfiguration : SkinConfiguration where TValue : class
         {
-            TValue val = null;
+            TValue val;
```

Comment:
[nit] You can hide `val` in next `if` something like `var val = query?.Invoke(conf)`

Reply:
Are you asking for something like that ?
```suggestion
            if ((source as Skin)?.Configuration is TConfiguration conf)
            {
                TValue val = query?.Invoke(conf);
                if (beatmapSkins && val != null)
                    return val;
            }
```

Reply:
Yeah, you can ignore my comment because surrounding code use your approach